### ISSUE TITLE
DPL-338 - Improve the way updated version numbers are enforced

### ISSUE THREAD

**Issue by sdjmchattie** [sanger/General-Backlog-Items#174]
**Description**
Merges to `master` on most repositories runs a check via GitHub Actions to identify whether the release version in the repository has been updated or not ready for the release to be distinctly labelled.  Sometimes this check fails because the Action is unable to identify what the last released version is.  This is because the step that identifies the most recent release is only looking at the past 30 releases and we do a lot of pre-releases.  If all 30 are pre-releases, no release version is found.  Then the check fails and must be performed manually.

Ideally the check would never fail and would always be able to determine that the release version has been updated.  Possible solutions offered are as follows:

* Some work was done on the action behind the step that checks the release version (https://github.com/pozetroninc/github-action-get-latest-release/pull/19).  The change provided was to make the action check the last 100 instead of 30 which is the maximum that Octokit would allow to be fetched at a time.  The action author would like us to add an attribute for the action to let the user decide to fetch a different number of releases rather than making 100 the default.  It's worth noting that this solution doesn't fix the problem if we then had more than 100 pre-releases.  Then pagination of the Octokit responses would be required.
* We could come up with a strategy for removing pre-release versions that are no longer relevant or not creating them in the first place.  It is currently very difficult to get a list of actual release version in one of the affected repositories.  It is overwhelmed by the number of pre-releases.  Most of the releases are not used for deployment and so are a wasteful use of resources.
* We could possibly find the previous release version by looking at the current file on `master` during the action and check that it was updated as part of the merge.  It might even be possible to look at the file list in the proposed merge and just ensure the release version file is amongst the files being changed.
* We could drop using semver for web apps.  i.e. we wouldn't have a release version file at all.  The version would be based on something else such as a timestamp or a reference to the GitHub action that built it.  Web app users don't get to choose which version they are using anyway and it only really holds value to PSD as a way of determining whether the affected version had the expected patches or not.

**Who the primary contacts are for this work**
Stuart Mc

**Knowledge or Stake holders**
James G, Thomas W

**Comment by BenTopping** [comment]
Another option could be to change the way we determine the release version. Could do something like adding a Major, Minor and Patch github label to a PR depending on the changes. Then read this label from the github action and automatically bump release depending on label. Could change the fail check to check one of these labels are attached instead of checking against last release name.

**Comment by sdjmchattie** [comment]
I've spent some time thinking about this and I think if we are going to do any work to the public GitHub action, it would be better if we just implemented the paging of releases until we find the most recent.  The reason being that even if we allow Octokit to return 100 at a time, that still doesn't guarantee finding the most recent release that meets the criteria.  So instead we should keep the results per page to 30 which is the Octokit default, and page through until we find the release.
